### DISCUSSION TITLE
dbc commit adds a bunch of 0x50 bytes to the output script

### DISCUSSION THREAD

**Discussion by adambor** [LNP-BP/nodes#7] [Support]
Following the demo here: https://github.com/LNP-BP/nodes/blob/master/contrib/demo-rgb.sh

After doing "dbc commit" a bunch of 0x50 are added to the output script which results in a provably unspendable output as can be seen here: https://mempool.space/testnet/tx/e006e0166ca407cb7492abdc9d1bfa3c72ab5cb8c06fa0f99afd00608748db90 . This also causes Tx ID to be different between than the one in the consignment (which is taken prior to dbc commit).

What should be the expected behavior here? I would expect "dbc commit" to actually produce the output script based on tapret commitment and default change output script (which in turn changes the TX ID - so shouldn't we finalize the consignment only after running dbc commit???).

I also tried running it without dbc commit, which resulted in the output script not being changed and was just a transaction from myself to myself (using classic change output script /1/0), so no tapret commitment in the output, but here the TX ID in consignment was also mismatched... Attaching the psbt, rgbc and raw tx for this case - [demo.zip](https://github.com/LNP-BP/nodes/files/9148573/demo.zip).
TX ID in consignment: 9324fe13ab0f61b1aa074eb7d549d008c4e22a01363d047447c51f8a17efce7d
Actual TX ID in psbt and raw tx: cc16cc0ce53360b61f5b1420ca1b2c28420040a153e3e0aa44b3bd7c0abb5ac0

There is definitely some issue with dbc commit, but maybe also `rgb-cli -n testnet transfer finalize --endseal ${TXOB} ${PSBT} ${CONSIGNMENT}` doesn't work as expected and doesn't take the correct TXID

**Comment by dr-orlovsky** [comment]
Yes, this is the intended behavior: `dbc commit` does tapret commit, which adds a provable unspendable branch to the script. This changes txid ideed - it is how tapret is working.

>  I would expect "dbc commit" to actually produce the output script based on tapret commitment and default change output script (which in turn changes the TX ID - so shouldn't we finalize the consignment only after running dbc commit???).

In this case the commitment will not be present in the final transaction, which will allow double spending/double commiting.

**Reply by louneskmt** [reply]
But, as the txid changes (`f3378fc1d125bc1b3f1340e68a315aa2b79cf26737b3becbf286239d114cf1be` vs `2da74e4b331f671a067b192835d66e96d109886bd519d52d9364808d81bc1b8f`), the commitment never gets to validation. It looks like the txid doesn't get updated in the commitment (`wbtg.rgbc`), so it is still looking for the old one. Or am I misunderstanding something?

```console
louneskmt@louneskmt-dedicated:~/RGB/instance1$ docker compose run lnpbp-tools btc-cold finalize --publish testnet wbtg.psbt                                                                                
02000000000101abc382a8a143a552d0827f0e85d318bac0f658768503a8bf10d17420e703baef0100000000ffffffff019411000000000000415050505050505050505050505050505050505050505050505050505050506a214bd08fbbfcb9a03d8aeb027cbae3576e3aace44793bd11554c0b17584f748d1f000141cb40fe2dfa387069302a3d00fc7a668d3c2fe0e25cdeba356bd7690a3a944338aed9874ee4006b95e839917bc2da2efae4aa8abd980fabbfd56da439701b18c50100000000                   

Connecting to network testnet using electrum.blockstream.info:60001
Transaction f3378fc1d125bc1b3f1340e68a315aa2b79cf26737b3becbf286239d114cf1be published

louneskmt@louneskmt-dedicated:~/RGB/instance1$ docker compose run lnpbp-tools rgb consignment validate wbtg.rgbc                                                                                           
---
unresolved_txids: []
unmined_endpoint_txids:
  - 2da74e4b331f671a067b192835d66e96d109886bd519d52d9364808d81bc1b8f
failures:
  - TransitionNotInAnchor:
      - 8ec90cdcd5c4e3634a230613db9ba2a6bc92248d979619c82784c25795481b5b
      - 2da74e4b331f671a067b192835d66e96d109886bd519d52d9364808d81bc1b8f
warnings:
  - EndpointTransactionMissed: 2da74e4b331f671a067b192835d66e96d109886bd519d52d9364808d81bc1b8f
info: []
```

**Reply by dr-orlovsky** [reply]
Not sure I exactly understand the issue. Inseed txid does change. Why this presents a problem? If you are trying to allocate the asset to the output of the witness transaction than you do not provide a txid and just an output number when creating an RGB state transition.

**Comment by oacdutra** [comment]
Hi @adambor and @louneskmt,

**Tapret Issue**
This issue was solved in the latest version of the RGB (v0.8.1). See details:
- https://github.com/BP-WG/bp-core/pull/24

**TX changes after finalize**
This issue has been fixied. See details:
- https://github.com/LNP-BP/client_side_validation/pull/103

Could you update the version and test again, please?

**Reply by louneskmt** [reply]
Hey @crisdut! Thanks a lot, I've seen the developments but haven't gotten the time to test them yet. Will do as soon as possible! 😁
